transforms/code/semantic_profiler/python/README.md

```diff
@@ -0,0 +1,73 @@
+# SP Transform 
+Please see the set of
+[transform project conventions](../../../README.md#transform-project-conventions)
+for details on general project conventions, transform configuration,
+testing and IDE set up.
+
+## Summary 
+This transform implements semantic profiling of a code dataset. Given an input dataset 
+as a pyarrow table with the UBSRs of code data points, this transform extracts the libraries
+and obtains their semantic mapping by consulting the IKB. The semantic concepts obatined per data
+point are then added as a new column in the input dataset. Those libraries which are not present in the
+IKB are recorded in a separate 'null_libs' file for offline processing. This file is passed as an input
+to the [offline path](src/offline_path/) which reads the libraries and obtains their semantic categories 
+from predefined set by prompting an LLM. The examples passed into the prompt are present in the [examples folder](src/examples/)
+
+## Configuration and command line Options
+
+The set of dictionary keys holding [SPTransform](src/sp_transform.py) 
+configuration for values are as follows:
+
+* _sp_ikb_file_ - This is the path to the IKB file which is a CSV file and by default located in the [IKB](src/ikb/) folder.
```

Comment:
Please follow this https://github.com/IBM/data-prep-kit/blob/dev/transforms/language/pdf2parquet/python/README.md and distinguish between transform configuration keys and cli parameters (the latter having the `sp_` prefix).

Reply:
In following the above template, please include the annotations that are added.